projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/memberships.php

```diff
@@ -246,6 +287,130 @@ public function create_products( $request ) {
 		return $request;
 	}
 
+	/**
+	 * Update an existing memberships product
+	 *
+	 * @param \WP_REST_Request $request The request passed from WP.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function update_product( \WP_REST_Request $request ) {
+		$product_id              = $request->get_param( 'product_id' );
+		$is_editable             = isset( $request['is_editable'] ) ? (bool) $request['is_editable'] : null;
+		$type                    = isset( $request['type'] ) ? $request['type'] : null;
+		$buyer_can_change_amount = isset( $request['buyer_can_change_amount'] ) && (bool) $request['buyer_can_change_amount'];
+
+		$payload = array(
+			'title'                   => $request['title'],
+			'price'                   => $request['price'],
+			'currency'                => $request['currency'],
+			'buyer_can_change_amount' => $buyer_can_change_amount,
+			'interval'                => $request['interval'],
+			'type'                    => $type,
+		);
+
+		if ( ( defined( 'IS_WPCOM' ) && IS_WPCOM ) ) {
+			require_lib( 'memberships' );
+			$product = Memberships_Product::get_from_post( get_current_blog_id(), $product_id );
+			if ( is_wp_error( $product ) ) {
+				return array( 'error' => $product->get_error_message() );
+			}
+			if ( ! $product || ! $product instanceof Memberships_Product ) {
+				return array( 'error' => __( 'Product not found.', 'jetpack' ) );
+			}
+
+			$updated_product = $product->update( $payload );
+			if ( is_wp_error( $updated_product ) ) {
+				return array( 'error' => $updated_product->get_error_message() );
+			}
+			return array( 'product' => $updated_product->to_array() );
+		} else {
+			$payload = array(
+				'type'     => $request['type'],
```

Comment:
This should probably have all the possible parameters (`title`, etc...) as well...